### PR TITLE
Fix the IE version number extractor to work without space between MSIE and the version number

### LINKED ISSUE #5858: Error in diaspora output: User Agent detection failed.
So I have several errors like this one in my console output running `script/server`

```
[2015-04-13T10:13:19.718770 #1529] ERROR -- : app error: undefined method `[]' for nil:NilClass (NoMethodError)
[2015-04-13T10:13:19.718901 #1529] ERROR -- : /home/diaspora/diaspora/lib/rack/chrome_frame.rb:68:in `ie_version'
[2015-04-13T10:13:19.718936 #1529] ERROR -- : /home/diaspora/diaspora/lib/rack/chrome_frame.rb:21:in `call'
[2015-04-13T10:13:19.718966 #1529] ERROR -- : /home/diaspora/.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/rack-1.6.0/lib/rack/deflater.rb:35:in `call'
[2015-04-13T10:13:19.718998 #1529] ERROR -- : /home/diaspora/diaspora/lib/unicorn_killer.rb:35:in `call'
[2015-04-13T10:13:19.719027 #1529] ERROR -- : /home/diaspora/.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/unicorn-4.8.3/lib/unicorn/http_server.rb:576:in `process_client'
[2015-04-13T10:13:19.719056 #1529] ERROR -- : /home/diaspora/.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/unicorn-4.8.3/lib/unicorn/http_server.rb:670:in `worker_loop'
[2015-04-13T10:13:19.719085 #1529] ERROR -- : /home/diaspora/.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/unicorn-4.8.3/lib/unicorn/http_server.rb:525:in `spawn_missing_workers'
[2015-04-13T10:13:19.719113 #1529] ERROR -- : /home/diaspora/.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/unicorn-4.8.3/lib/unicorn/http_server.rb:536:in `maintain_worker_count'
[2015-04-13T10:13:19.719142 #1529] ERROR -- : /home/diaspora/.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/unicorn-4.8.3/lib/unicorn/http_server.rb:294:in `join'
[2015-04-13T10:13:19.719170 #1529] ERROR -- : /home/diaspora/.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/unicorn-4.8.3/bin/unicorn_rails:209:in `<top (required)>'
[2015-04-13T10:13:19.719199 #1529] ERROR -- : /home/diaspora/.rbenv/versions/2.1.5/bin/unicorn_rails:23:in `load'
[2015-04-13T10:13:19.719226 #1529] ERROR -- : /home/diaspora/.rbenv/versions/2.1.5/bin/unicorn_rails:23:in `<main>'
```

I checked [line 68](https://github.com/diaspora/diaspora/blob/develop/lib/rack/chrome_frame.rb#L68) with Jonne, it appears that `ua_string.match(/MSIE (\S+)/)` returns `nil`. That means, no space after `MSIE`.

Here is the evil User Agent I found in the nginx logs: `Mozilla/4.0 (compatible; MSIE8.0; Windows NT 6.0) .NET CLR 2.0.50727` (Berk, Vista).

To solve this issue, one needs to write a spec testing this user agent and then correct the regex. I see if I find some time to do it. 

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/11125513-error-in-diaspora-output-user-agent-detection-failed?utm_campaign=plugin&utm_content=tracker%2F522&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F522&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Flaburgan**
Close #5858

I'd like to add a test to see if `ie_version()` returns the right number but I don't know how to call this method from the spec.

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -756,6 +755,7 @@ DEPENDENCIES
   omniauth-twitter (= 1.0.1)
   omniauth-wordpress (= 0.2.1)
   open_graph_reader (= 0.5.0)
+  pg (= 0.18.1)
   pry
   pry-byebug
   pry-debundle
```
Comment (jhass) [inline Gemfile.lock]: Please do not commit this change to Gemfile.lock

File: Gemfile.lock [hunk 2]
```diff
@@ -786,7 +786,7 @@ DEPENDENCIES
   rails_admin (= 0.6.7)
   rb-fsevent (= 0.9.4)
   rb-inotify (= 0.9.5)
-  redcarpet (= 3.2.3)
+  redcarpet (= 3.2.2)
   remotipart (= 1.2.1)
   responders (= 2.1.0)
   roxml (= 3.1.6)
```
Comment (jhass) [inline Gemfile.lock]: Looks like the reset went wrong there, run `git fetch upstream` and retry.

**Comment by Flaburgan** [comment]
Now running for ~48h on my pod, no problem so far and no more unicorn crash in the output :)

**Comment by jhass** [comment]
Thanks!

**Pull request merged**